File: src/main/java/com/hedvig/backoffice/services/tickets/TicketService.java
package com.hedvig.backoffice.services.tickets;

import com.hedvig.backoffice.domain.QuestionGroup;
import com.hedvig.backoffice.graphql.types.RemindNotification;
import com.hedvig.backoffice.services.tickets.dto.CreateTicketDto;
import com.hedvig.backoffice.services.tickets.dto.FullTicketHistoryDto;
import com.hedvig.backoffice.services.tickets.dto.TicketDto;
import com.hedvig.backoffice.services.tickets.dto.TicketStatus;

import java.util.List;
import java.util.UUID;

public interface TicketService {

  List<TicketDto> getAllTickets(Boolean resolved);

  TicketDto getTicketById(UUID ticketId);

  FullTicketHistoryDto getTicketWithFullHistory(UUID ticketId);

  UUID createNewTicket(CreateTicketDto ticket, String createdBy);

  void changeDescription(UUID ticketId, String newDescription, String modifiedBy);

  void assignToTeamMember(UUID ticketId, String teamMemberId, String modifiedBy);

  void changeStatus(UUID ticketId, TicketStatus newStatus, String modifiedBy);

  void changeReminder(UUID ticketId, RemindNotification newReminder, String modifiedBy);

  void changePriority(UUID ticketId, float newPriority, String modifiedBy);

  void createTicketFromQuestions (QuestionGroup questionGroup);

  void setQuestionGroupAsDone(String groupId, String changedBy );


}

File: src/main/java/com/hedvig/backoffice/services/tickets/TicketService.kt
package com.hedvig.backoffice.services.tickets

import com.hedvig.backoffice.domain.QuestionGroup
import com.hedvig.backoffice.graphql.types.RemindNotification
import com.hedvig.backoffice.services.tickets.dto.CreateTicketDto
import com.hedvig.backoffice.services.tickets.dto.TicketHistoryDto
import com.hedvig.backoffice.services.tickets.dto.TicketDto
import com.hedvig.backoffice.services.tickets.dto.TicketStatus
import java.util.UUID

interface TicketService {

    fun getAllTickets(resolved: Boolean?): List<TicketDto>

    fun getTicketById(ticketId: UUID): TicketDto

    fun getTicketHistory(ticketId: UUID): TicketHistoryDto

    fun createTicket(ticket: CreateTicketDto, createdBy: String): UUID

    fun changeDescription(ticketId: UUID, newDescription: String, modifiedBy: String)

    fun changeAssignedTo(ticketId: UUID, assignedTo: String, modifiedBy: String)

    fun changeStatus(ticketId: UUID, newStatus: TicketStatus, modifiedBy: String)

    fun changeReminder(ticketId: UUID, newReminder: RemindNotification, modifiedBy: String)

    fun changePriority(ticketId: UUID, newPriority: Float, modifiedBy: String)
}
